railties/lib/rails/commands/unused_routes/unused_routes_command.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+require "rails/command"
+
+module Rails
+  module Command
+    class UnusedRoutesCommand < Base # :nodoc:
+      class RouteInfo
+        def initialize(route)
+          requirements = route.requirements
+          @controller_name = requirements[:controller]
+          @action_name = requirements[:action]
+          @controller_class = (@controller_name.to_s.camelize + "Controller").safe_constantize
+        end
+
+        def invalid?
+          controller_class_missing? || (action_missing? && template_missing?)
+        end
+
+        private
+          def view_path(root)
+            File.join(root.path, @controller_name, @action_name)
+          end
+
+          def controller_class_missing?
+            @controller_name && @controller_class.nil?
+          end
+
+          def template_missing?
+            @controller_class && @controller_class.try(:view_paths).to_a.flat_map { |path| Dir["#{view_path(path)}.*"] }.none?
+          end
+
+          def action_missing?
+            @controller_class && @controller_class.instance_methods.exclude?(@action_name.to_sym)
+          end
+      end
+
+      def perform(*)
+        require_application_and_environment!
+        require "action_dispatch/routing/inspector"
+
+        if invalid_routes.any?
+          say("Found #{invalid_routes.count} invalid #{"route".pluralize(invalid_routes.count)}:")
+          say("")
+          say(inspector.format(formatter))
+        else
+          say("No invalid routes found.")
+        end
```

Comment:
```suggestion
        if invalid_routes.any?
          say("Found #{invalid_routes.count} invalid #{"route".pluralize(invalid_routes.count)}:")
          say("")
          say(inspector.format(formatter))
          exit(1)
        else
          say("No invalid routes found.")
        end
```

There might be a better way to do this, but it would be neat if you could use this to fail CI builds.

Reply:
I had this in the original script, but I think the test runner didn't like to nonzero return. I'll look into how I can address this. I can probably change how Rails shells out for railtie tests. 👍